ObjectsAsAPI/Models/CancelOrderRequest.cs

```diff
@@ -0,0 +1,122 @@
+﻿using MongoDB.Bson;
+using ObjectsAsAPI.Services;
+using Realms;
+
+namespace ObjectsAsAPI.Models;
+
+public partial class CancelOrderRequest : IRealmObject, IRequest<CancelOrderPayload, CancelOrderResponse>
+{
+    [PrimaryKey]
+    [MapTo("_id")]
+    public ObjectId Id { get; set; } = ObjectId.GenerateNewId();
+
+    [MapTo("_creatorId")]
+    public string CreatorId { get; set; }
+
+    [MapTo("status")]
+    private string _Status { get; set; } = null!;
+
+    public RequestStatus Status
+    {
+        get => Enum.Parse<RequestStatus>(_Status);
+        set => _Status = value.ToString();
+    }
+
+    [MapTo("createdAt")]
+    public DateTimeOffset CreatedAt { get; set; }
+
+    [MapTo("payload")]
+    public CancelOrderPayload? Payload { get; set; }
+
+    [MapTo("response")]
+    public CancelOrderResponse? Response { get; set; }
+
+    // Used in the UI
+    public string? Description
+    {
+        get
+        {
+            string? status = null;
+
+            var requestType = "CancelOrder";
+            var orderIdentifier = Payload?.OrderId.ToString();
+
+            if (orderIdentifier?.Length > 10)
+            {
+                orderIdentifier = orderIdentifier[..10];
+            }
+
+            if (Response != RealmValue.Null)
+            {
+                status = Response?.Status switch
+                {
+                    ResponseStatus.Approved => "✅ ",
+                    ResponseStatus.Rejected => "❌ ",
+                    _ => throw new NotImplementedException(),
+                };
+            }
+
+            return $"{status}{requestType}{(string.IsNullOrEmpty(orderIdentifier) ? "" : $" - {orderIdentifier}")}";
+        }
+    }
+
+    // Used in the UI
+    public string? StatusString
+    {
+        get
+        {
+            return Status switch
+            {
+                RequestStatus.Draft => "Draft",
+                RequestStatus.Pending => "Pending",
+                RequestStatus.Handled => $"{Response?.Status}{(string.IsNullOrEmpty(Response?.RejectedReason) ? "" : $": \"{Response?.RejectedReason}\"")}",
+                _ => throw new NotImplementedException(),
+            };
+        }
+    }
```

Comment:
Yes, it could be handled by a value converter, I wanted to make this simpler because we need a `MultiValueConveter` for these I think. It looks kinda ugly as it is now though. 

Regarding the notifications, that's what I'm trying to do with `OnPropertyChanged` down here

Reply:
While we don't need to design this in a way that's super robust and future-proof, the current implementation of `OnPropertyChanged` is somewhat heavily tied to the business logic, where we make the assumption that `Response` only changes when `Status` changes. Technically, it's possible that `Response` or `Response.RejectedReason` change independently of `Status` in which case `StatusString` will not emit notifications. That being said, if a multivalueconverter further complicates things, I'd be fine with leaving the implementation as it is and just adding a comment in `OnPropertyChanged` for why checking for `Status` is sufficient.

Reply:
Agree with that